android_sample/app/src/main/jni/main.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018 The Amber Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <android/log.h>
+#include <android_native_app_glue.h>
+
+#include "amber/amber.h"
+#include "amber/recipe.h"
+#include "amber/result.h"
+#include "amber_script.h"
+
+// Android log function wrappers
+static const char* kTAG = "Amber";
+#define LOGE(...) \
+  ((void)__android_log_print(ANDROID_LOG_ERROR, kTAG, __VA_ARGS__))
+
+static void amber_main(android_app* app) {
+  amber::android::AmberScriptLoader loader(app);
+
+  amber::Result r = loader.LoadAllScriptsFromAsset();
+  if (!r.IsSuccess()) {
+    LOGE("%s", r.Error().c_str());
+    return;
+  }
+
+  const auto& script_info = loader.GetScriptInfo();
+
+  for (const auto& info : script_info) {
+    amber::Amber am;
+    amber::Recipe recipe;
+    amber::Result r = am.Parse(info.script_content, &recipe);
+    if (!r.IsSuccess()) {
+      LOGE("%s:\n\t%s", info.file_name.c_str(), r.Error().c_str());
+      return;
+    }
+
+    amber::Options amber_options;
+    r = am.ExecuteWithShaderData(&recipe, amber_options, info.shader_map);
+    if (!r.IsSuccess()) {
+      LOGE("%s:\n\t%s", info.file_name.c_str(), r.Error().c_str());
+      return;
```

Comment:
On failure, I think it's better to try to continue.  That way if we try 5 scripts, 1 fails early then we see 4 pass, 1 fail.